### PR TITLE
New version of the Avahi package with a redesigned interface.

### PR DESCRIPTION
**Pull request by dennypage**
A completely new version of the Avahi package.

The new version offers the following improvements:

- A much cleaner and simpler interface.
- Uses positive interface selection rather than negative selection. This prevents security surprises when adding interfaces.
- Addresses an information leakage issue with OpenVPN connections. 
- Removes run-time dependance upon dbus, reducing memory and improving service startup time (no more 5 second delay).
- Disables local cache which caused hostname collision issues with reflection.

This package has been tested with 2.4.3 (prod) and 2.4.4 (dev).

Thanks,
Denny

### CODE REVIEW

**Comment by dennypage** [comment]
I've now tested with 2.4.4 release. All seems good with the package itself, but I did notice something unusual. With the package installed via the command line interface (pkg install pfSense-pkg-Avahi-2.0.0.txz), the package no longer shows up in the list of installed packages in the UI. In 2.4.3-p1, it would show up with an exclamation mark next to the version number indicating that the version installed was newer than the version in the package repo. Is this intended? Is this something I need to fix?

**Comment by jim-p** [comment]
That is normal for packages that didn't come from pfSense. See the release blog post, release notes, and upgrade guide which all have a big warning about how packages from other repositories behave now on 2.4.4.

**Comment by dennypage** [comment]
"Packages from alternate repositories will not appear in the Installed Packages list in the GUI, and must be entirely managed in the command line."

Sorry Jim, I just didn't read down far enough. My bad.

File: net/pfSense-pkg-Avahi/files/etc/inc/priv/avahi.priv.inc [hunk 1]
```diff
@@ -3,7 +3,7 @@
  * avahi.priv.inc
  *
  * part of pfSense (https://www.pfsense.org)
- * Copyright (c) 2015 Rubicon Communications, LLC (Netgate)
+ * Copyright (c) 2018 Denny Page
```
Comment (rbgarga) [inline net/pfSense-pkg-Avahi/files/etc/inc/priv/avahi.priv.inc]: Don't remove Rubicon Copyright

File: net/pfSense-pkg-Avahi/files/usr/local/pkg/avahi.xml [hunk 2]
```diff
@@ -3,12 +3,12 @@
...
 <packagegui>
 	<copyright>
-<![CDATA[
+	<![CDATA[
 /*
  * avahi.xml
  *
  * part of pfSense (https://www.pfsense.org)
- * Copyright (c) 2009-2015 Rubicon Communications, LLC (Netgate)
+ * Copyright (c) 2018 Denny Page
```
Comment (rbgarga) [inline net/pfSense-pkg-Avahi/files/usr/local/pkg/avahi.xml]: Don't remove Rubicon Copyright

File: net/pfSense-pkg-Avahi/files/usr/local/pkg/avahi/avahi.inc [hunk 3]
```diff
@@ -0,0 +1,128 @@
+<?php
+/*
+ * avahi.inc
+ *
+ * part of pfSense (https://www.pfsense.org)
+ * Copyright (c) 2018 Denny Page
```
Comment (rbgarga) [inline net/pfSense-pkg-Avahi/files/usr/local/pkg/avahi/avahi.inc]: Don't remove Rubicon Copyright

File: net/pfSense-pkg-Avahi/pkg-plist [hunk 4]
```diff
@@ -1,7 +1,6 @@
 pkg/avahi.xml
-pkg/avahi.inc
+pkg/avahi/avahi.inc
```
Comment (rbgarga) [inline net/pfSense-pkg-Avahi/pkg-plist]: Why move .inc file to a separate directory?

File: net/pfSense-pkg-Avahi/files/etc/inc/priv/avahi.priv.inc (hunk 1 above)
Comment (dennypage) [reply net/pfSense-pkg-Avahi/files/etc/inc/priv/avahi.priv.inc]: I didn't remove the copyright--in effect it wasn't there to begin with. This package is a complete ground up writing, unrelated to the prior package. This particular file was initialized by copying and modifying the lldld.priv.inc.

File: net/pfSense-pkg-Avahi/files/usr/local/pkg/avahi.xml (hunk 2 above)
Comment (dennypage) [reply net/pfSense-pkg-Avahi/files/usr/local/pkg/avahi.xml]: I didn't remove the copyright--in effect it wasn't there to begin with. This package is a complete ground up writing, unrelated to the prior package. This particular file was initialized by copying and modifying lldld.xml.

File: net/pfSense-pkg-Avahi/files/usr/local/pkg/avahi/avahi.inc (hunk 3 above)
Comment (dennypage) [reply net/pfSense-pkg-Avahi/files/usr/local/pkg/avahi/avahi.inc]: I didn't remove the copyright--in effect it wasn't there to begin with. This package is a complete ground up writing, unrelated to the prior package. This particular file was written completely from scratch.

File: net/pfSense-pkg-Avahi/pkg-plist (hunk 4 above)
Comment (dennypage) [reply net/pfSense-pkg-Avahi/pkg-plist]: Same pattern used in other packages, which allows the introduction of additional files in the future. Same as with the nut and lldpd packages.

**Pull request merged**